build.sh

```
@@ -156,9 +157,17 @@ START_BUILDS=$(now)
 
 cd /mnt/raspotify
 
-# Get the git rev of raspotify for .deb versioning
-RASPOTIFY_GIT_VER="$(git describe --tags "$(git rev-list --tags --max-count=1)" 2>/dev/null || echo unknown)"
-RASPOTIFY_HASH="$(git rev-parse HEAD | cut -c 1-7 2>/dev/null || echo unknown)"
+echo 'Obtaining latest Git repository tag for DEB package version ...'
+RASPOTIFY_GIT_VER="$(git describe --tags "$(git rev-list --tags --max-count=1)" || :)"
+if [ -z "$RASPOTIFY_GIT_VER" ]; then
+	echo 'Could not obtain latest tag from local repository. Obtaining it from upstream: https://api.github.com/repos/dtcooper/raspotify/tags'
```

How about starting the whole subroutine with `git fetch --tags origin` instead? This will ensure tags locally and not trigger a HTTP request to GitHub.

---

The problem is not that tags are locally missing, but that the fork repo in fact does not have any tags, also not online. I could rephrase the log to "local/origin", to make that clear. It just makes sense for anyone who contributes to the project, forking the repo to make an edit, then open a PR upstream: forks do not contain any tags, unless they are created at the fork.

---

I see. But is it really necessary to have the correct tag output in the build unless it is a release build? What is the value of having this tag?

---

Very well, however I fear it will potentially silently ignore tags from a (faulty) local copy.
I want the checked out copy to be the source of truth for the version tag, not the upstream.

---

That would be best, but the only thing which contains the version string in a local copy/fork without tags is the `CHANGELOG.md`, which has not a very machine-readable format. I.e. the projects version solely relies on Git tags.

The version could be defined somewhere in the code, but then it needs to be additionaly maintained. The latest Git tag could still override the hardcoded version, to maintain current behavior for upstream releases.

---

Hi again.
How about a guard before L163 that checks whether the locally checked-out copy's origin is `dtcooper/raspotify`? This way we can have the best of both worlds.

---

Not so trivial to know whether the local repo is supposed to be `dtcooper/raspotify` or not. Git itself has no concept for owner or repo name, this is all a GitHub-only thing. Best I could find is checking the `origin` remote URL, which is by default matching repository that was originally cloned. GitHub allows some variations for this URL, and even that they are all redirected to a canonical, `git remote` still returns the variant that was entered when cloning the repo.

... okay and it does not work for the pull request triggered runs: On my fork, the push triggered builds were successful, but here not. Reason is that the origin indeed is `https://github.com/dtcooper/raspotify`, and the "branch"/"ref" reflects the code from my fork instead. So we would need to check the ref as well, whether it is a pull request (should be possible to see from the name scheme). But this is becoming quite complicated, worth the hassle?

---

> But this is becoming quite complicated and error-prone/not failsafe, worth the hassle?

Screw it then -- it was worth a shot, but I prefer simplicity.

I'll try merging this PR as-is and see where that gets us :)

In the future I want to implement _nightly builds_, i.e. the latest commit on `main` should produce a GitHub pre-release where users always can download the latest debs, for testing.

---

I'll merge through the GitHub clickops after you revert that last commit then.